### ISSUE TITLE
Waiting on field event to stream file

### ISSUE THREAD

**Issue by halljared** [mscdex/busboy#32]
Hi, my use case is that I'm specifying a particular location where I would like to upload a file. The location is in the same form as the file. In order to do this, I'm using a promise that forces the file event handler to wait on the field event handler. See this code:

``` javascript
console.debug('Responding to POST request to ./views/uploadFile');
var promisedFolder = new Deferred(),
    responseObj = {files:[]};

var busboy = new Busboy({ headers: req.headers });
busboy.on('file', function(fieldname, file, filename, encoding, mimetype) {
    console.debug('File [' + fieldname + ']: filename: ' + filename);
    promisedFolder.then(function(folder) {
        console.debug('Got folder from request, writing file');
        var fullPath = path.normalize('/' + folder + '/' + filename);
        file.pipe(fs.createWriteStream(fullPath));
        file.on('error', function(err) {
            console.debug('Got error streaming file');
        });
        responseObj.files.push({name:filename, url:'#'});
    });
});
busboy.on('field', function(fieldname, val, valTruncated, keyTruncated) {
    console.debug('Field [' + fieldname + ']: value: ' + val);
    promisedFolder.resolve(val);
});
busboy.on('finish', function() {
    console.debug('Done parsing form!');
    res.send(responseObj);
});

req.pipe(busboy);
```

This is what I get in the logs, but then nothing else happens.

```
DEBUG: 2014-03-13T16:14:43.212Z: File [files[]]: filename: jare-2008-articlespdf.pdf
DEBUG: 2014-03-13T16:14:43.212Z: Field [path]: value: /data/Bills/
DEBUG: 2014-03-13T16:14:43.212Z: Got folder from request, writing file
```

Weirdly, If I resend the exact same request, the upload goes through.

Thoughts?
